Azure/sonic-buildimage
# [build]: Move Systemd service start to systemd generator

**- What I did**
* Move the enabling of Systemd services from sonic_debian_extension to a new systemd generator

**- How I did it**
* Create a new systemd generator to manually create symlinks to enable systemd services
* Add rules/Makefile to build generator
* Add services to be enabled to `/etc/sonic/generated_services.conf` to be read by the generator at boot time

**- How to verify it**
Build, install, and start an image and ensure that all system services and docker containers run properly.

Output of `docker ps` after boot should be similar to the following:

```
admin@sonic:~$ docker ps
CONTAINER ID        IMAGE                             COMMAND                  CREATED             STATUS              PORTS               NAMES
c51475ca0653        docker-syncd-vs:latest            "/usr/bin/supervisord"   12 seconds ago      Up 11 seconds                           syncd
232e01bf384e        docker-dhcp-relay:latest          "/usr/bin/docker_ini…"   13 seconds ago      Up 11 seconds                           dhcp_relay
1a2b1fe59130        docker-teamd:latest               "/usr/bin/supervisord"   15 seconds ago      Up 13 seconds                           teamd
d4d144ef119c        docker-snmp-sv2:latest            "/usr/bin/supervisord"   15 seconds ago      Up 13 seconds                           snmp
a13a558645c8        docker-router-advertiser:latest   "/usr/bin/supervisord"   15 seconds ago      Up 13 seconds                           radv
88a23838dce0        docker-orchagent:latest           "/usr/bin/supervisord"   18 seconds ago      Up 16 seconds                           swss
065467d78f3c        docker-platform-monitor:latest    "/usr/bin/docker_ini…"   23 seconds ago      Up 22 seconds                           pmon
83868140ee1c        docker-lldp-sv2:latest            "/usr/bin/supervisord"   23 seconds ago      Up 22 seconds                           lldp
9b7c603859a7        docker-fpm-frr:latest             "/usr/bin/supervisord"   23 seconds ago      Up 22 seconds                           bgp
49c44f5b3139        docker-sonic-telemetry:latest     "/usr/bin/supervisord"   23 seconds ago      Up 22 seconds                           telemetry
7a8da33a617f        docker-database:latest            "/usr/bin/supervisord"   3 hours ago         Up 27 seconds                           database
```

All containers should be running or 'up'.
**- Description for the changelog**

Enable Systemd services in a Systemd generator to prepare for boot time platform detection and service configuration.

**- A picture of a cute animal (not mandatory but encouraged)**
